app/controllers/document_lead_image_controller.rb

```diff
@@ -117,6 +117,30 @@ def choose_image
     redirect_to document_path(document)
   end
 
+  def delete_image
+    document = Document.find_by_param(params[:document_id])
+    image = document.images.find(params[:image_id])
+
+    begin
+      if image.id == document.lead_image_id
+        DocumentUpdateService.update!(
+          document: document,
+          user: current_user,
+          type: "lead_image_removed",
+          attributes_to_update: { lead_image_id: nil },
+        )
+      end
+
+      AssetManagerService.new.delete(image)
+      image.destroy
```

Comment:
I guess we have a somewhat annoying situation here where if the request to asset manager fails you remove the item as a lead image but don't remove the image from your document. Could consider putting this in a transaction so it's an all or nothing? (though I know it's not great practice to do a http request while in a transaction)

Reply:
So this behaviour is intended (the Asset Manager down spec covers it), but I get it might not be what the user wants. As long as it leaves everything in a valid state and the user understands that their action may not have completely worked, I think it's reasonable to leave it as it is for now to avoid the additional complexity - we'd have to do another Pub API request to get the state consistent again.